### ISSUE TITLE
picarto.tv changed the design of the site

### ISSUE THREAD

**Issue by Shinkir0** [ytdl-org/youtube-dl#28532]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.03.14**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-f', 'bestvideo+bestaudio/best', '-o', 'S:/Downloads/%(extractor)s/%(uploader)s/%(title)s.%(ext)s', '-a-', '-v']
[debug] Batch file urls: ['https://picarto.tv/Monipue']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2021.03.25
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: ffmpeg N-100625-gf6477ac9f4, ffprobe N-100625-gf6477ac9f4
[debug] Proxy map: {}
[Picarto] Monipue: Downloading JSON metadata
[Picarto] Monipue: Downloading load balancing info
ERROR: Unable to download JSON metadata: HTTP Error 405: Not Allowed (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\common.py", line 634, in _request_webpage
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\YoutubeDL.py", line 2279, in urlopen
  File "C:\Python\Python34\lib\urllib\request.py", line 470, in open
  File "C:\Python\Python34\lib\urllib\request.py", line 580, in http_response
  File "C:\Python\Python34\lib\urllib\request.py", line 508, in error
  File "C:\Python\Python34\lib\urllib\request.py", line 442, in _call_chain
  File "C:\Python\Python34\lib\urllib\request.py", line 588, in http_error_default

## Description
Site changed design, Now on the Stremer page, you can not copy the link to the video, as well as if the broadcast is currently going it is not saved. For example here https://picarto.tv/Monipue

**Comment by tERB00** [comment]
Can confirm same issue

**Comment by Shinkir0** [comment]
I would like to be able to download playlist.
As far as I noticed the template for the link to the video and the tumbnail are very similar, that's a feature we can use in some way

https://picarto.tv/Zombi3Guts

https://recording-eu-1.picarto.tv//thumbs/golive/z/zo/Zombi3Guts/golive+Zombi3Guts_2021.02.03.03.38.31.mkv.jpg
https://recording-eu-1.picarto.tv//thumbs/golive/z/zo/Zombi3Guts/golive+Zombi3Guts_2021.01.18.09.21.52_nsfw.mkv.jpg
https://recording-eu-1.picarto.tv//thumbs/golive/z/zo/Zombi3Guts/golive+Zombi3Guts_2021.01.18.09.21.28.mkv.jpg
https://recording-eu-1.picarto.tv//thumbs/golive/z/zo/Zombi3Guts/golive+Zombi3Guts_2021.01.10.05.52.34.mkv.jpg
https://recording-eu-1.picarto.tv//thumbs/golive/z/zo/Zombi3Guts/golive+Zombi3Guts_2021.01.09.05.48.16.mkv.jpg

https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+Zombi3Guts_2021.02.03.03.38.31.mkv/index.m3u8
https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+Zombi3Guts_2021.01.18.09.21.52_nsfw.mkv/index.m3u8
https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+Zombi3Guts_2021.01.18.09.21.28.mkv/index.m3u8
https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+Zombi3Guts_2021.01.10.05.52.34.mkv/index.m3u8
https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+Zombi3Guts_2021.01.09.05.48.16.mkv/index.m3u8

**Comment by Gho57X90** [comment]
Was going to open an issue about this, but found this one that's appears obviously relevant. Found out about the _"index.m3u8"_ myself working as a roundabout way to still achieve the same results with a little more effort, but today I've found it unfortunately does not seem to work anymore?

The following is what happened when I tried to download a Picarto offline video via a batch script I wrote up for ease. As far as I can tell, there's nothing wrong with the script itself, so I _think_ Picarto changed something during one of their recent instances of maintenance.

### Verbose log
-------------------------
```
D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>title youtube-dl - DEDICATED INSTANCE 01

D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>cd C:\youtube-dl

D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>youtube-dl.exe --verbose --no-part --hls-prefer-ffmpeg https://recording-eu-1.picarto.tv/stream/hls/govodbgolive+Wouhlven_2021.04.17.07.26.15_nsfw.mkv/1_0/index.m3u8?sessId=28003
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '--no-part', '--hls-prefer-ffmpeg', 'https://recording-eu-1.picarto.tv/stream/hls/govodbgolive+Wouhlven_2021.04.17.07.26.15_nsfw.mkv/1_0/index.m3u8?sessId=28003']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2021.03.25
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: ffmpeg git-2020-07-24-21442a8
[debug] Proxy map: {}
[generic] index: Requesting header
[generic] index: Downloading m3u8 information
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\YoutubeDL.py", line 806, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\YoutubeDL.py", line 827, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\common.py", line 534, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\generic.py", line 2446, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\common.py", line 1369, in _sort_formats
youtube_dl.utils.ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

Script complete! This program will now end.
Press any key to continue . . .
```
In short; downloading from Picarto seems to be fully broken now.

I would like to ask if a patch could be made that would make downloading with this roundabout method workable again until further R&D could be done. If so, that would be really appreciated. It _did_ work, so it would suffice if that's all that can be done right now.

As an end-user, not sure what there is to be done about it otherwise though.

**Comment by tERB00** [comment]
> As an end-user, not sure what there is to be done about it otherwise though.

Your URI is wrong. It should be like this: https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+MarTheDog_2021.04.18.19.22.39_nsfw.mkv/index.m3u8?token=YOUR_TOKEN

You can get this link by clicking "debug info" on the media player.

**Comment by melodicstream** [comment]
> Your URI is wrong. It should be like this: https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+MarTheDog_2021.04.18.19.22.39_nsfw.mkv/index.m3u8?token=YOUR_TOKEN
> 
> You can get this link by clicking "debug info" on the media player.

Thank you so much. I can confirm that this works. I was trying a lot of different URLs (the popout one, the one without it being from the recording servers, etc). Never occurred to me to try out that debug URL. Such a very small detail, but it helped me out a lot.

Now the only thing left is for Picarto to allow me to select and copy the URL in the debug info. Right now I need to Inspect Element on it to be able to copy it, and it's annoying. Unfortunately, not a youtube-dl issue.

**Comment by tERB00** [comment]
@melodicstream It wont allow you to select it on the bottom but if you scroll that little text window you can find same URI there and copy it.

**Comment by Gho57X90** [comment]
> Your URI is wrong. It should be like this: https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+MarTheDog_2021.04.18.19.22.39_nsfw.mkv/index.m3u8?token=YOUR_TOKEN

I checked again to make sure the URL was right, and while you're correct that my originally used URL did not match up with your example, I still got the same results even after fixing it.

```
D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>title youtube-dl - DEDICATED INSTANCE 01

D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>cd C:\youtube-dl

D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>youtube-dl.exe --verbose --no-part --hls-prefer-ffmpeg https://recording-eu-1.picarto.tv/stream/hls/govodbgolive+Wouhlven_2021.04.17.07.26.15_nsfw.mkv/1_0/index.m3u8?sessId=9978
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '--no-part', '--hls-prefer-ffmpeg', 'https://recording-eu-1.picarto.tv/stream/hls/govodbgolive+Wouhlven_2021.04.17.07.26.15_nsfw.mkv/1_0/index.m3u8?sessId=9978']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2021.03.25
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: ffmpeg git-2020-07-24-21442a8
[debug] Proxy map: {}
[generic] index: Requesting header
[generic] index: Downloading m3u8 information
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\YoutubeDL.py", line 806, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\YoutubeDL.py", line 827, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\common.py", line 534, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\generic.py", line 2446, in _real_extract
  ```

Using the Stream URI from the debug info section of the player yielded no better results.

```
D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>title youtube-dl - DEDICATED INSTANCE 01

D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>cd C:\youtube-dl

D:\[GENERAL STORAGE]\Picarto Offline VODs\youtube-dl\INSTANCE 01>youtube-dl.exe --verbose --no-part --hls-prefer-ffmpeg https://recording-eu-1.picarto.tv/stream/hls/govodbgolive+Wouhlven_2021.04.17.07.26.15_nsfw.mkv/index.m3u8?token=NTIyMjU4XzIyNjQwX3JhbmR1bWJsYWk=
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '--no-part', '--hls-prefer-ffmpeg', 'https://recording-eu-1.picarto.tv/stream/hls/govodbgolive+Wouhlven_2021.04.17.07.26.15_nsfw.mkv/index.m3u8?token=NTIyMjU4XzIyNjQwX3JhbmR1bWJsYWk=']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2021.03.25
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: ffmpeg git-2020-07-24-21442a8
[debug] Proxy map: {}
[generic] index: Requesting header
[generic] index: Downloading m3u8 information
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\YoutubeDL.py", line 806, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\YoutubeDL.py", line 827, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\common.py", line 534, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\generic.py", line 2446, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp5nc448kk\build\youtube_dl\extractor\common.py", line 1369, in _sort_formats
youtube_dl.utils.ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

Script complete! This program will now end.
Press any key to continue . . .```

**Comment by tERB00** [comment]
@Gho57X90 You are missing "%2B" in the govodbgolive string. It should be "govod%2bgolive". I think you somehow lost these symbols while copying and pasting. I've just checked "https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+Wouhlven_2021.04.17.07.26.15_nsfw.mkv/index.m3u8?token=NTIyMjU4XzIyNjQwX3JhbmR1bWJsYWk=" and it is downloading fine.

**Comment by Gho57X90** [comment]
> You are missing "%2B" in the govodbgolive string. It should be "govod%2bgolive". I think you somehow lost these symbols while copying and pasting.

It doesn't seem like that's the case on my end, though I wish it was as that would make the issue stupidly simple. This is the literal script I'm using right now.
```
title youtube-dl - DEDICATED INSTANCE 01
cd C:\youtube-dl
youtube-dl.exe --verbose --no-part --hls-prefer-ffmpeg https://recording-eu-1.picarto.tv/stream/hls/govod%2bgolive+Wouhlven_2021.04.17.07.26.15_nsfw.mkv/index.m3u8?token=NTIyMjU4XzIyNjQwX3JhbmR1bWJsYWk=
::@color 27
@echo Script complete! This program will now end.
@pause
```
![notepad++_2021-04-22_15-12-15](https://user-images.githubusercontent.com/11093715/115658958-22803880-a37d-11eb-8f22-0f34299b3b9f.png)

**Comment by tERB00** [comment]
@Gho57X90 Oh - I see what's your issue now. In windows batch files "%2" means second parameter you passed to the .bat file from command line. Since it's empty %2 being replaced with nothing resulting "govodbgolive"

**Comment by Gho57X90** [comment]
> @Gho57X90 Oh - I see what's your issue now. In windows batch files "%2" means second parameter you passed to the .bat file from command line. Since it's empty %2 being replaced with nothing resulting "govodbgolive"

Ya-hah! Yes, that's the problem, it seems. So basically, you gotta replace the _"%2b"_ with a plus symbol.
![image](https://user-images.githubusercontent.com/11093715/115660448-607e5c00-a37f-11eb-8a3e-0502cacf76d4.png)

Looks to be working now like you said. _(Thanks for the help with that, @tERB00.)_
c:
![image](https://user-images.githubusercontent.com/11093715/115660269-1dbc8400-a37f-11eb-9a81-c9e50b18a906.png)

Would still be really good to find a better way to handle Picarto offline videos moving forward, but this works again in the meantime.

**Comment by Shinkir0** [comment]
Token copying not necessarily without it also downloads
Also the downloadable file always applies the same name "index [index] .mp4" because of which it will not be possible to simultaneously download multiple files, You need to write a unique file name each time.
I use such a script
`
@echo off
:: Data entry:
set /p tile="tile:"
paste | youtube-dl -f "bestvideo+bestaudio/best" -o "s:/Downloads/PicartoVod/NA/%tile% - %%(title)s.%%(ext)s" -a- -v
`
for him, you need https://gist.github.com/jpflouret/19da43372e643352a1bf#file-paste-zip
1. copy the link to the clipboard
2. I start the batnik
3. you are prompted to enter a name, (I take the name from the link)
4. I press enter
the paste program itself passes the link from the buffer to the youtube-dl input

**Comment by Laybert** [comment]
Anyone care to troubleshoot the issue I've have downloading offline Picarto videos since the UI switch? Been getting the same load of errors with the current command lines recommended in this thread, and current versions of youtube-dl (in and out of a clean folder). This hasn't affected most other video sites I've downloaded from.

```
E:\youtube-dl>youtube-dl --verbose --no-part --hls-prefer-ffmpeg https://recording-eu-1.picarto.tv/stream/hls/govod+golive+Griefgoose_2021.05.17.18.59.44.mkv/index.m3u8
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '--no-part', '--hls-prefer-ffmpeg', 'https://recording-eu-1.picarto.tv/stream/hls/govod+golive+Griefgoose_2021.05.17.18.59.44.mkv/index.m3u8']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2021.06.06
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: none
[debug] Proxy map: {}
[generic] index: Requesting header
[generic] index: Downloading m3u8 information
[debug] Default format spec: best/bestvideo+bestaudio
[debug] Invoking downloader on 'https://recording-eu-1.picarto.tv/stream/hls/govod+golive+Griefgoose_2021.05.17.18.59.44.mkv/1_0/index.m3u8?sessId=22739'
[download] Destination: index-index.mp4
ERROR: m3u8 download detected but ffmpeg or avconv could not be found. Please install one.
  File "__main__.py", line 19, in <module>
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\__init__.py", line 475, in main
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\__init__.py", line 465, in _real_main
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 2069, in download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 808, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 815, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 847, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 881, in process_ie_result
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 1692, in process_video_result
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 1976, in process_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 1915, in dl
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\downloader\common.py", line 366, in download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\downloader\external.py", line 35, in real_download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\downloader\external.py", line 227, in _call_downloader
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\downloader\common.py", line 165, in report_error
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 628, in report_error
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 590, in trouble
```
